src/manifest/mod.ts

```diff
@@ -180,20 +180,30 @@ export class SlackManifest {
     this.definition.types.push(customType);
   }
 
+  /**
+   * Verifies scopes defined in the app passes baseline validation.
+   * @returns {string[]} The user-defined manifest scopes from `definition.botScopes`
+   */
   private ensureBotScopes(): string[] {
-    const includedScopes = this.definition.botScopes || [];
-
-    // Add datastore scopes if necessary
+    // Warn about missing datastore scopes if app includes datastores
     if (Object.keys(this.definition.datastores ?? {}).length > 0) {
+      const missingScopes: string[] = [];
       const datastoreScopes = ["datastore:read", "datastore:write"];
```

Comment:
I know this is not part of the changes but should this be placed is some sort of a global constant location instead of  directly hard coded in the function?

Reply:
Yeah I thought about that too, but then it is only used here in this one place, and the performance impact of this is negligible, so I figured it would make the code more readable to keep the array near where it is used. Open to changing it if we think that is worth it.